utils-intervalrdd/src/main/scala/org/bdgenomics/utils/interval/array/IntervalArray.scala

```diff
@@ -108,6 +108,7 @@ object IntervalArray extends Serializable {
 trait IntervalArray[K <: Interval[K], T] extends Serializable {
   val array: Array[(K, T)]
   val maxIntervalWidth: Long
+  var optLastIndex: Option[Int] = None
```

Comment:
This should be private.

Also, this variable won't be thread safe, which is slightly problematic for how we're using it from Spark. However, it has a trivial workaround: you do a shallow copy of the IntervalArray. I think we should:

1. Document that this change makes a single instance of the IntervalArray thread unsafe
2. Document that the simple workaround is to make a shallow copy of the array

Also, I'm usually a big `Option` stickler, but I think we should do some performance profiling to see what the performance hit is for using `Option[Int]` vs an `Int` whose default value is `null` here.